src/UIKit/UIGestureRecognizer.cs

```diff
@@ -11,19 +11,18 @@
 #if !WATCH
 
 using System;
-using System.Collections;
 using System.Collections.Generic;
 using Foundation; 
 using ObjCRuntime;
-using CoreGraphics;
+#nullable enable
 
 namespace UIKit {
 
 	public partial class UIGestureRecognizer {
 		//
 		// Tracks the targets (NSObject, which we always enforce to be Token) to the Selector the point to, used when disposing
 		//
-		Dictionary<Token,IntPtr> recognizers = new Dictionary<Token,IntPtr> ();
+		Dictionary<Token,IntPtr>? recognizers = new Dictionary<Token,IntPtr> ();
```

Comment:
I think it might be easier/better to keep `recognizers` as non-nullable (remove the `?`), and then in the Dispose method clear the dictionary instead of assigning null:

```cs
recognizers.Clear ();
```

That shouldn't change logic much (the dictionary will be freed by the GC a little bit later, and that's it), but most importantly it'll make some of the other changes you've made unnecessary (see below).